### DISCUSSION TITLE
Hosting a Clog app on Google Cloud Run (serverless)

### DISCUSSION THREAD

**Discussion by iantor** [rabbibotton/clog#233] [General]
I am curious if it would be a good idea to host a Clog app on Google Cloud Run. They have a free tier which is interesting, and it supports Docker images, so I could use an SBCL docker image with Clog and app incorporated.  However, there are some considerations when using Websockets on GCR.  On the free tier, the "server" is only alive when serving HTTP requests, and Websocket operations are included in this category.  However, if the websocket remains connected, then CPU billing would occur.  Does Clog have long-running websocket connections, or are they just connected for brief periods?

[official docs](https://cloud.google.com/run/docs/triggering/websockets)

**Comment by rabbibotton** [comment]
The current implementation uses a websocket that remains in place. GNOGA that clog is based on can be set to use Ajax but even in that case it keeps open a long poll connection. I am willing to figure out other solutions. How would a page on their hosting handle push notifications?

**Reply by iantor** [reply]
I'm not sure, but based on what you've said I don't think that Google Cloud Run would be a good fit.  I would probably use Amazon Lightsail as I have in the past.  At the base level it can be as low as $3.50 a month flat rate, and also supports Docker images.  Then there will be no issues with websockets.  Unlike Digital Ocean, AWS-hosted sites are also accessible from China, which is another interest of mine.  (unless they violate some arbitrary rules, but these can be understood and followed).
